---
title: "DNA methylation at quantitative trait loci (mQTLs) varies with cell type and non-heritable factors and may improve breast cancer risk assessment "
author: "Main Data"
date: ""
output:
  pdf_document:
    toc: false
    latex_engine: xelatex
header-includes:
  \usepackage{fontspec}
  \usepackage[utf8]{inputenc}
  \usepackage{pdfpages}
  \usepackage{graphicx}
  \setmainfont{Arial}
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = F, warning = F, message = F)
# library(dplyr)
# library(tidyverse)
# library(patchwork)
# library(ggplot2)
# library(here)
# library(ggsci)
# library(gt)
# library(gtsummary)
# library(here)
# 
# setwd(here())
# # Load scripts
# scripts <- list.files("0-source-code", full.names = T)
# for (s in scripts) {source(s)}
# 
# # Set themes and colours
# theme_set(theme(panel.background = element_blank())) # ggplot panel
# col <- pal_npg()(8) # cols
```

Chiara Herzog, Allison Jones, Iona Evans, Michal Zikan, David Cibula, Nadia Harbeck, Nicoletta Colombo, Angelique Flöter Rådestad, Kristina Gemzell-Danielsson, Nora Pashayan and Martin Widschwendter

\tableofcontents

\pagebreak

\section{Main items}


```{r fig1}
# # Figure 1C ---------
# # variability of top Breast variable CpGs in cervical, buccal, and blood samples.
# load("1-analysis-pipeline/1-output/variability_breast_matched.Rdata")
# variability <- variability |>
#   dplyr::mutate(set = readr::parse_number(set),
#                 tissue = paste0(tissue, " sample"))
# 
# fig1c <- variability |>
#   dplyr::mutate(tissue = factor(tissue, levels = c("cervical sample", "buccal sample", "blood sample"))) |>
#   ggplot(aes(x = set,
#              y = median)) +
#   geom_line(aes(linetype = icgroup,
#                 colour = tissue),
#             linewidth = 0.75) +
#   ylim(0, 0.17)  +
#   scale_linetype_manual(values = c("solid", "dotted", "twodash", "dashed", "longdash", "dashed", "dotted"),
#                         labels = c("All samples", "< 25% ic", "25-50% ic", "50-75% ic", "> 75% ic", ">50% myeloid", "<50% myeloid"),
#                         breaks = c("sd_all", "sd_0_25", "sd_25_50",
#                                    "sd_50_75", "sd_75_100",
#                                    "sd_0_50", "sd_50_100"),
#                         name = "") +
#   theme_bw() +
#   theme(legend.background = element_blank(),
#         axis.text.x = element_text(angle = 60),
#         legend.key = element_blank(),
#         # legend.key.size = unit(15, "mm"),
#         legend.key.width = unit(16, "mm")) +
#   scale_colour_manual(values = col[c(5,2,1)],
#                       name = "") +
#   scale_x_continuous(breaks = c(1,2,5,10,15,25),
#                      labels = c("1%", "2%", "5%", "10%", "15%", "20%")) +
#   theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
#   xlab("Top variable breast CpGs") +
#   ylab("Median standard deviation") +
#   facet_wrap(~tissue)
# 
# grDevices::cairo_pdf("2-markdown/0-figure-panels/fig1c.pdf",
#                      width = 10, height = 3.9)
# print(fig1c)
# dev.off()
# 
# # # Figure 1D --------------------
# # Variability of mQTL and all CpGs
# load("1-analysis-pipeline/1-output/sd_tissues.Rdata") # load data
# 
# # Get mQTL CpGs
# mqtl <- readxl::read_xlsx("1-analysis-pipeline/0-data/10549_2021_6185_MOESM1_ESM.xlsx", sheet = 6, skip = 3) |>  # get mQTL CpGs
#   janitor::clean_names() |>
#   pull(cp_g_site)
# intersect <- intersect(mqtl, sd_tissues$cervical$cg)
# mqtl <- mqtl[match(intersect, mqtl)]
# 
# # Prep data: cervical
# sd_cpgs_all_cervical<- sd_tissues$cervical |>
#   dplyr::mutate(set = 'all')
# sd_cpgs_mqtl_cervical <- sd_tissues$cervical |>
#   filter(cg %in% mqtl) |>
#   dplyr::mutate(set = 'mQTL')
# sd_cpg_cervical <- rbind(sd_cpgs_all_cervical, sd_cpgs_mqtl_cervical)
# sd_long_cervical <- sd_cpg_cervical |>
#   pivot_longer(c(2:6),
#                names_to = "icgroup",
#                values_to = "sd") |>
#   mutate(icgroup = case_when(icgroup == "sd_all" ~ "All samples",
#                              icgroup == "sd_0_25" ~ "< 25% ic",
#                              icgroup == "sd_25_50" ~ "25-50% ic",
#                              icgroup == "sd_50_75"~ "50-75% ic",
#                              icgroup == "sd_75_100" ~ "> 75% ic")) |>
#   mutate(icgroup = factor(icgroup, levels = c("All samples", "< 25% ic", "25-50% ic", "50-75% ic", "> 75% ic"))) |>
#   mutate(set = factor(set, levels = c("mQTL", "all")),
#          tissue = "cervical sample")
# 
# # Buccal sample
# sd_cpgs_all_buccal<- sd_tissues$buccal |>
#   dplyr::mutate(set = 'all')
# sd_cpgs_mqtl_buccal <- sd_tissues$buccal |>
#   filter(cg %in% mqtl) |>
#   dplyr::mutate(set = 'mQTL')
# sd_cpg_buccal <- rbind(sd_cpgs_all_buccal, sd_cpgs_mqtl_buccal)
# sd_long_buccal <- sd_cpg_buccal |>
#   pivot_longer(c(2:6),
#                names_to = "icgroup",
#                values_to = "sd") |>
#   mutate(icgroup = case_when(icgroup == "sd_all" ~ "All samples",
#                              icgroup == "sd_0_25" ~ "< 25% ic",
#                              icgroup == "sd_25_50" ~ "25-50% ic",
#                              icgroup == "sd_50_75"~ "50-75% ic",
#                              icgroup == "sd_75_100" ~ "> 75% ic")) |>
#   mutate(icgroup = factor(icgroup, levels = c("All samples", "< 25% ic", "25-50% ic", "50-75% ic", "> 75% ic"))) |>
#   mutate(set = factor(set, levels = c("mQTL", "all")),
#          tissue = "buccal sample")
# 
# # Blood sample
# sd_cpgs_all_blood<- sd_tissues$blood |>
#   dplyr::mutate(set = 'all')
# sd_cpgs_mqtl_blood <- sd_tissues$blood |>
#   filter(cg %in% mqtl) |>
#   dplyr::mutate(set = 'mQTL')
# sd_cpg_blood <- rbind(sd_cpgs_all_blood, sd_cpgs_mqtl_blood)
# sd_long_blood <- sd_cpg_blood |>
#   pivot_longer(c(2:4),
#                names_to = "icgroup",
#                values_to = "sd") |>
#   mutate(icgroup = case_when(icgroup == "sd_all" ~ "All samples",
#                              icgroup == "sd_0_50" ~ "< 50% myeloid",
#                              icgroup == "sd_50_100" ~ "> 50% myeloid")) |>
#   mutate(icgroup = factor(icgroup, levels = c("All samples", "< 50% myeloid", "> 50% myeloid"))) |>
#   mutate(set = factor(set, levels = c("mQTL", "all")),
#          tissue = "blood sample")
# 
# sd_long<- rbind(sd_long_buccal, sd_long_cervical, sd_long_blood) |>
#   dplyr::mutate(tissue = factor(tissue, levels = c("cervical sample", "buccal sample", "blood sample")))
# 
# median1 <- median(sd_cpgs_mqtl_cervical$sd_all)
# median2 <- median(sd_long_cervical[sd_long_cervical$set=="mQTL",]$sd)
# 
# 
# ## Tests: within tissue - mQTL v all
# df <- sd_long |>
#   dplyr::filter(icgroup == "All samples") |>
#   group_by(tissue) |>
#   dplyr::summarise(pval = signif(wilcox.test(sd[set=="mQTL"], sd[set=="all"])$p.value, 2)) |>
#   mutate(padj = p.adjust(pval, method = "bonferroni", n = 6))
# 
# # cervical mQTL v all: 6.0e-63
# # buccal mQTL v all: 1.8e-48
# # blood mQTL v all: 9.0e-116
# 
# # Compare between tissues
# 
# cerv_buccal <- wilcox.test(sd_long[sd_long$tissue == "cervical sample" & sd_long$set == "mQTL" & sd_long$icgroup == "All samples",]$sd, sd_long[sd_long$tissue == "buccal sample" & sd_long$set == "mQTL" & sd_long$icgroup == "All samples",]$sd, paired = TRUE)$p.value
# 
# cerv_blood <- cerv_buccal <- wilcox.test(sd_long[sd_long$tissue == "cervical sample" & sd_long$set == "mQTL" & sd_long$icgroup == "All samples",]$sd, sd_long[sd_long$tissue == "blood sample" & sd_long$set == "mQTL" & sd_long$icgroup == "All samples",]$sd, paired = TRUE)$p.value
# 
# buccal_blood <- cerv_buccal <- wilcox.test(sd_long[sd_long$tissue == "blood sample" & sd_long$set == "mQTL" & sd_long$icgroup == "All samples",]$sd, sd_long[sd_long$tissue == "buccal sample" & sd_long$set == "mQTL" & sd_long$icgroup == "All samples",]$sd, paired = TRUE)$p.value
# 
# x <- p.adjust(c(cerv_buccal, cerv_blood, buccal_blood), n = 6, method = "bonferroni")
# # cervical v buccal: 1.662789e-30
# # cervical v blood: 8.196439e-71
# # blood v buccal 1.662789e-30
# 
# tmp1 <- sd_long |>
#   dplyr::filter(set == "all") |>
#   dplyr::group_by(tissue, icgroup) |>
#   dplyr::summarise(sdmean = mean(sd)) |>
#   dplyr::filter(icgroup %in% c("All samples", ", 25 % ic"))
# 
# sd_long <- sd_long |>
#   dplyr::left_join(tmp1, by = c("tissue", "icgroup")) |>
#   ungroup()
# 
# 
# fig1d <- sd_long |>
#   ggplot(aes(x = set,
#              y = sd,
#              fill = interaction(icgroup, tissue))) +
#   geom_boxplot(outlier.size = 0.1) +
#   scale_fill_manual(values=c(col[5], alpha(col[5], alpha = 0.8), alpha(col[5], alpha = 0.6), alpha(col[5], alpha = 0.4),  alpha(col[5], alpha = 0.2),
#                              col[2], alpha(col[2], alpha = 0.8), alpha(col[2], alpha = 0.6), alpha(col[2], alpha = 0.4), alpha(col[2], alpha = 0.2),
#                              col[1], alpha(col[1], alpha = 0.8), alpha(col[1], alpha = 0.6)),
#                     name = "") +
#   facet_wrap(~tissue) +
#   geom_hline(aes(yintercept = median1), linetype = "dashed") +
#   geom_hline(aes(yintercept = median2), linetype = "dotted") +
#   theme_bw() +
#   xlab("") + ylab("standard deviation of CpGs")
# 
# # annotate across facets in illustrator
# grDevices::cairo_pdf("2-markdown/0-figure-panels/fig1d.pdf",
#                      width = 10, height = 3.9)
# print(fig1d)
# dev.off()
```

\includegraphics[trim=0 9cm 0 0]{0-figure-panels/fig1.pdf}

\subsection{Figure 1. Identification of the most informative surrogate sample for breast-variable DNA methylation indicates that cervical samples exhibit higher variability in the top-variable breast CpGs compared to matched buccal and blood samples.} \textbf{a} Higher variability indicates the potential presence of more information, opposed to CpGs which are homogeneously methylated or unmethylated across samples. We explored this by identifying variability of CpGs in tissue at risk of breast cancer (=breast tissue) and assessing variability of these top breast-variable CpGs in three non-invasive surrogate tissues, utilising matched buccal, blood, and cervical samples from the same individuals (n=222 per tissue). \textbf{b} Standard deviation of the top variable breast CpGs (1, 2, 5, 10, 15, and 20 percentiles, respectively) in matched cervical, buccal, and blood samples (all ,and separated by inferred immune cell composition). \textbf{c} Variability of all CpGs versus mQTL CpGs in the three matched tissues (all or separated by inferred immune cell composition). The dashed line shows median variability of all CpGs in all cervical samples while the solid line shows median variability of mQTLs in cervical samples with an immune cell composition (ic) < 25%. 

Boxplot boxes indicate interquartile range. Abbreviations: DNAme, DNA methylation. mQTL, methylation quantitative trait locus. SNP, single nucleotide polymorphism. ic, immune cell. 



\newpage

```{r fig2}
# # Fig2a: Cervical samples -> breast v buccal ---------------------
# load("1-analysis-pipeline/0-data/pheno.Rdata")
# 
# pheno <- pheno |>
#   filter(dataset == "validation set") |>
#   mutate(type = case_when(type == "Control" ~ "Control",
#                           type == "Breast cancer" ~ "Breast\ncancer")) |>
#   mutate(type = factor(type, levels = c("Control", "Breast\ncancer"))) |>
#   droplevels()
# 
# test <- signif(t.test(pheno[pheno$type=="Control",]$index_mQTL, pheno[pheno$type == "Breast\ncancer",]$index_mQTL)$p.value,2)
# 
# df2 <- pheno |>
#   group_by(type) |>
#   dplyr::summarise(n = paste(type, "\n(n = ", n(), ")", sep = "")) |>
#   distinct()
# df2 <- unique(df2)
# 
# labs <- "cervical sample"
# names(labs) <- "validation set"
# 
# # # AUC for manuscript
# # pROC::ci(pheno$type, pheno$index_mQTL)
# # ind1 <- (pheno$ER=="Neg" & !is.na(pheno$ER)) | pheno$type == "Control"
# # ind2 <- (pheno$ER=="Pos" & !is.na(pheno$ER)) | pheno$type == "Control"
# # pROC::auc(pheno[ind1,]$type, pheno[ind1,]$index_mQTL)
# # pROC::ci(pheno[ind1,]$type, pheno[ind1,]$index_mQTL)
# # pROC::auc(pheno[ind2,]$type, pheno[ind2,]$index_mQTL)
# # pROC::ci(pheno[ind2,]$type, pheno[ind2,]$index_mQTL)
# 
# fig2a <- pheno |>
#   ggplot(aes(x = type,
#              y = index_mQTL)) +
#   geom_hline(yintercept = 0,
#              linetype = "dashed",
#              colour = "gray60",
#              size = 0.5) +
#   geom_violin(aes(fill = type),
#               alpha = 0.2) +
#   geom_boxplot(width = 0.1) +
#   ggbeeswarm::geom_beeswarm(aes(colour = type),
#                             size = 0.6,
#                             alpha = 0.7) +
#   scale_fill_manual(values = col[c(3,4)],
#                     aesthetics = c("colour", "fill")) +
#   theme(legend.position = "none") +
#   facet_wrap(~dataset, labeller = labeller(dataset = labs)) +
#   annotate("segment", x = 1, xend = 2,
#            y = 3.5, yend = 3.5,
#            size = 0.5,
#            colour = "gray40") +
#   annotate("text",
#            label = "p < 0.001",
#            size = 3,
#            x = 1.5,
#            y = 3.9,
#            colour = "gray40") +
#   xlab("") +
#   ylab("WID™-qtBC") +
#   scale_x_discrete(label = df2$n)
# 
# # grDevices::cairo_pdf("2-markdown/0-figure-panels/fig2a.pdf",
# #                      width = 4, height = 3.9)
# # print(fig2a)
# # dev.off()
# 
# # Fig 2b: ER AUC (cervical) ------------------------
# source("0-source-code/plot_two_rocs_ER.R")
# fig2b <- plot_two_rocs_ER(pheno$type, pheno$index_mQTL, pheno$ER)
# 
# # grDevices::cairo_pdf("2-markdown/0-figure-panels/fig2b.pdf",
# #                      width = 3.9, height = 3.9)
# # print(fig2b)
# # dev.off()
# 
# # Fig 2c (permutation) ------------------------
# load("1-analysis-pipeline/3-output/permutation.Rdata")
# 
# ci <- round(Rmisc::CI(df$auc,
#                       ci = 0.95),2)
# dens.fun <- function(z) with (density(df$value),approx(x,y,z)$y)
# 
# densplot <- df |>
#   ggplot() +
#   geom_density(aes(x = auc),
#                bw = 0.005,
#                size = 0.4,
#                colour = "gray40") +
#   theme(panel.background = element_blank()) +
#   scale_x_continuous(n.breaks = 10,
#                      limits = c(0,1)) +
#   xlab("AUC (validation set)") +
#   ylab("Density")
# 
# dpb <- ggplot_build(densplot)
# x1 <- min(which(dpb$data[[1]]$x >=.62))
# x2 <- max(which(dpb$data[[1]]$x <=.63))
# 
# fig2c <- densplot +
#   geom_area(data=data.frame(x=dpb$data[[1]]$x[x1:x2],
#                             y=dpb$data[[1]]$y[x1:x2]),
#             aes(x=x, y=y), fill=col[6], alpha = 0.4) +
#   coord_cartesian(xlim=c(0.45, 0.8)) +
#   annotate("segment",
#            x = 0.5, xend = 0.5,
#            y = 0, yend = 5.5,
#            colour = "gray60",
#            size = 0.5,
#            linetype = "dashed") +
#   annotate("text",
#            x = 0.5,
#            y = 7,
#            size = 2.9,
#            label = "Expected\nat random",
#            colour = "gray60") +
#   annotate("text",
#            x = 0.62,
#            y = 21,
#            size = 2.9,
#            colour = "grey40",
#            label = "Permutation\nn = 10000") +
#   annotate("text",
#            x = 0.71,
#            y = 7,
#            size = 2.9,
#            label = "WID™-qtBC\nindex",
#            colour = "black") +
#   annotate("segment",
#            x = 0.71, xend = 0.71,
#            y = 0, yend = 5.5,
#            colour = "black",
#            size = 0.5) +
#   annotate("segment",
#            x = mean(df$auc), xend = mean(df$auc),
#            y = 0, yend = 18.5,
#            colour = "grey40",
#            size = 0.5)
# 
# # grDevices::cairo_pdf("2-markdown/0-figure-panels/fig2c.pdf",
# #                      width = 3.9, height = 3.9)
# # print(fig2c)
# # dev.off()
# 
# # Fig 2d (breast tissue) ------------------------
# load("1-analysis-pipeline/0-data/pheno.Rdata")
# 
# pheno <- pheno |>
#   filter(dataset == "breast validation set") |>
#   mutate(type = factor(type, levels = c("Normal", "Normal-adjacent"))) |>
#   droplevels()
# 
# p1 <- wilcox.test(pheno[pheno$type == "Normal",]$index_mQTL, pheno[pheno$type == "Normal-adjacent",]$index_mQTL)$p.value
# 
# df <- pheno |>
#   group_by(type) |>
#   dplyr::summarise(n = paste(type, "\n(n = ", n(), ")", sep = "")) |>
#   distinct()
# 
# labs <- "breast tissue"
# names(labs) <- "breast validation set"
# 
# fig2d <- pheno |>
#   ggplot(aes(x = type,
#              y = index_mQTL)) +
#   geom_violin(aes(fill = type),
#               alpha = 0.2) +
#   geom_boxplot(width = 0.1) +
#   ggbeeswarm::geom_beeswarm(aes(colour = type),
#                             size = 1,
#                             alpha = 0.7) +
#   scale_fill_manual(values = col[c(3,6)],
#                     aesthetics = c("fill", "colour")) +
#   theme(legend.position = "none") +
#   xlab("") +
#   ylab("WID™-qtBC") +
#   scale_x_discrete(label = df$n) +
#   annotate("text", x = 1.5, y = 0.5, label = paste("p = ", round(p1,2), sep = ""), size = 3.2, colour = "gray40") +
#   annotate("segment", x = 1, xend = 2, y = 0.25, yend = 0.25, size = 0.5, colour = "gray40") +
#   facet_wrap(~dataset, labeller = labeller(dataset = labs))
# 
# # grDevices::cairo_pdf("2-markdown/0-figure-panels/fig2d.pdf",
# #                      width = 4, height = 3.9)
# # print(fig2d)
# # dev.off()
# #
# # # Fig 2e (aucs breast) ------------------------
# load("1-analysis-pipeline/0-data/pheno.Rdata")
# pheno <- pheno |>
#   filter(dataset == "breast validation set") |>
#   dplyr::mutate(type = ifelse(type == "Normal", "Control", type)) |>
#   mutate(type = factor(type, levels = c("Control", "Normal-adjacent"))) |>
#   droplevels()
# 
# fig2e <- plot_roc(pheno$type, pheno$index_mQTL, size = 2.9) +
#   theme(panel.grid = element_blank())
# 
# # grDevices::cairo_pdf("2-markdown/0-figure-panels/fig2e.pdf",
# #                      width = 3.9, height = 3.9)
# # print(fig2e)
# # dev.off()
# 
# 
# plot<- (fig2a|fig2b|fig2c)/(fig2d|fig2e|plot_spacer()) + plot_annotation(tag_levels = "a")
# 
# grDevices::cairo_pdf("2-markdown/0-figure-panels/fig2.pdf",
#                      width = 10, height = 8)
# print(plot)
# dev.off()
```

\includegraphics[trim=0 8cm 0 0]{0-figure-panels/fig2.pdf}

\subsection{Figure 2. The WID™-qtBC index distinguishes breast cancer cases and controls in cervical and breast samples.} \textbf{a} Outline of classifier development and validation strategy. \textbf{b} The WID™-qtBC index is increased in cervical samples from current breast cancer cases compared to controls in the validation set (p=1.2e-07 in two-tailed Student’s t-test). \textbf{c} The WID™-qtBC has an AUC of 0.71 and is unaffected by estrogen receptor (ER) status. \textbf{d} Permutation analysis of index training featuring randomly selected CpGs indicates that the WID™-qtBC AUC in the validation set is higher than expected by chance. Shaded area indicates 95% confidence interval from permutation testing. \textbf{e} The WID™-qtBC index is increased in normal tissue adjacent to breast cancer compared to normal breast tissue. \textbf{f} ROC curve for discrimination of normal to normal-adjacent tissue based on the WID™-qtBC index.

Boxplot boxes indicate interquartile range. Abbreviations: AUC, area under the receiver operating characteristic curve. ER-, estrogen receptor negative breast cancer. ER+, estrogen receptor positive breast cancer.

\newpage

```{r fig3}
# # Fig 3A: Correlation with PRS -------------
# load("1-analysis-pipeline/0-data/pheno.Rdata")
# dat <- pheno |>
#   dplyr::filter(dataset == "validation set" & !is.na(polygenic_risk_score_bc)) |>
#   droplevels() |>
#   mutate(type = factor(type, levels = c("Control", "Breast cancer"))) |>
#   mutate(grp = case_when(polygenic_risk_score_bc > median(polygenic_risk_score_bc) ~ "high PRS",
#                        polygenic_risk_score_bc <= median(polygenic_risk_score_bc) ~ "low PRS")) |>
#   mutate(grp = factor(grp, levels = c("low PRS", "high PRS")))
# 
# fig3a <- dat |>
#   ggplot(aes(x = polygenic_risk_score_bc,
#              y = index_mQTL,
#              colour = type)) +
#   geom_point(size = 0.75,
#              alpha = 0.8) +
#   geom_smooth(method = "lm",
#               se = FALSE) +
#   theme(legend.position = "top",
#         panel.background = element_blank(),
#         legend.key = element_blank()) +
#   xlab("PRS₃₁₃") +
#   ylab("WID™-qtBC") +
#   scale_colour_manual(values = col[c(3,4)],
#                       name = "") +
#   ggpubr::stat_cor(size = 3)
# 
# # grDevices::cairo_pdf("2-markdown/0-figure-panels/fig3a.pdf",
# #                      width = 3.9, height = 3.9)
# # print(fig3a)
# # dev.off()
# 
# # Fig 3B: AUC-------------
# fig3b <- plot_roc_dichot(dat$type, dat$index_mQTL, dat$grp,
#                      title1 = "all",
#                      title2 = "low PRS₃₁₃",
#                      title3 = "high PRS₃₁₃")
# 
# # grDevices::cairo_pdf("2-markdown/0-figure-panels/fig3b.pdf",
# #                      width = 3.9, height = 3.9)
# # print(fig3b)
# # dev.off()
# 
# # Fig 3C Odds ratio: --------------------
# load("1-analysis-pipeline/0-data/pheno.Rdata")
# dat <- pheno |>
#   filter(dataset == "validation set" & !is.na(polygenic_risk_score_bc)) |>
#   mutate(type = factor(type, levels = c("Control", "Breast cancer"))) |>
#   droplevels()
# 
# prs <- ortable_percentiles(dat$polygenic_risk_score_bc, dat$type, c(25, 50, 75, 100),
#                           "PRS313")
# qtbc <- ortable_percentiles(dat$index_mQTL, dat$type, c(25, 50, 75, 100),
#                            "WID-qtBC")
# # combined - both z scored and summed up.
# dat2 <- dat |>
#   mutate(prs_z = scale(polygenic_risk_score_bc),
#          qtbc_z = scale(index_mQTL),
#          score = prs_z+qtbc_z)
# combined <- ortable_percentiles(dat2$score, dat2$type, c(25, 50, 75, 100),
#                                 "combined")
# 
# # Merge data
# data <- rbind(prs, qtbc, combined)
# data$scoretype <- gsub("PRS313", "PRS₃₁₃", data$scoretype)
# data$scoretype <- gsub("WID-qtBC", "WID™-qtBC", data$scoretype)
# data$p <- as.numeric(data$p)
# 
# fig3c <- data |>
#   ggplot(aes(x = percentile,
#              y = log2(or),
#              group = scoretype)) +
#   geom_line(aes(colour = scoretype)) +
#   geom_ribbon(aes(ymin = log2(ci_lo),
#                   ymax = log2(ci_hi),
#                   fill = scoretype),
#               alpha = 0.2) +
#   xlab("Percentile categories for risk score") + ylab("Odds ratio\n(logarithmic scale)") +
#   scale_colour_manual(values = c(col[8], "grey50", "black"),
#                       aesthetics = c("fill", "colour")) +
#   theme(legend.position = "top",
#         legend.title = element_blank(),
#         panel.background = element_blank(),
#         panel.grid = element_blank())
# 
# # Fig 3D PRS/Stages --------------------
# load("1-analysis-pipeline/0-data/pheno.Rdata")
# 
# # Subset dataset
# dat <- pheno |>
#   filter(dataset == "validation set" & !is.na(polygenic_risk_score_bc) & type == "Breast cancer") |>
#   droplevels()
# 
# # Create groups
# dat <- dat |>
#   mutate(grp = case_when(polygenic_risk_score_bc > median(polygenic_risk_score_bc) ~ "high PRS",
#                          polygenic_risk_score_bc <= median(polygenic_risk_score_bc) ~ "low PRS"),
#          grp = factor(grp, levels = c("low PRS", "high PRS")),
#          qtgrp = case_when(index_mQTL > median(index_mQTL) ~ "high WID-qtBC",
#                            index_mQTL <= median(index_mQTL) ~ "low WID-qtBC"),
#          qtgrp = factor(qtgrp, levels = c("low WID-qtBC", "high WID-qtBC"))) |>
#   tidyr::unite("group", grp:qtgrp, sep = ",\n", remove = FALSE) |>
#   mutate(group = factor(group, levels = c("low PRS,\nlow WID-qtBC",
#                                           "high PRS,\nlow WID-qtBC",
#                                           "low PRS,\nhigh WID-qtBC",
#                                           "high PRS,\nhigh WID-qtBC")))
# # Reorder factors
# dat <- dat |>
#   mutate(ER = case_when(ER == "Pos" ~ "ER+",
#                         ER == "Neg" ~ "ER-",
#                         is.na(ER) ~ "Unknown"),
#          PR = case_when(PR == "Pos" ~ "PR+",
#                         PR == "Neg" ~ "PR-",
#                         is.na(PR) ~ "Unknown"),
#          stage_N = case_when(stage_N == "N0" ~ "N0",
#                              stage_N %in% c("N1", "N2", "N3") ~ "N1-3",
#                              stage_N == "NX" | is.na(stage_N) ~ "Unknown"),
#          stage_T = case_when(stage_T %in% c("T1", "Tis") ~ "T1",
#                              stage_T %in% c("T2", "T3", "T4") ~ "T2-4",
#                              is.na(stage_T) ~ "Unknown"),
#          grade = case_when(grade %in% c("Grade I", "Grade II") ~ "Grade I/II",
#                            grade == "Grade III" ~ "Grade III"),
#          HER2 = case_when(HER2 == "Pos" ~ "HER2+",
#                           HER2 == "Neg" ~ "HER2-",
#                           is.na(HER2) ~ "Unknown")) |>
#   mutate(ER = factor(ER, levels = c("ER-", "ER+", "Unknown")),
#          PR = factor(PR, levels = c("PR+", "PR-", "Unknown")),
#          stage_N = factor(stage_N, levels = c("N1-3", "N0", "Unknown")),
#          stage_T = factor(stage_T, levels = c("T2-4", "T1", "Unknown")),
#          grade = factor(grade, levels = c("Grade III", "Grade I/II")),
#          HER2 = factor(HER2, levels = c("HER2+", "HER2-", "Unknown")))
# 
# # Reorder for plotting
# dat2 <- dat |>
#   pivot_longer(c(ER, PR, HER2, stage_N, stage_T, grade),
#                names_to = "set",
#                values_to = "value") |>
#   mutate(set = factor(set,
#                       levels = c("ER", "PR", "HER2", "stage_N", "stage_T", "grade"),
#                       labels = c("ER status", "PR status", "HER2 status", "Nodal stage",
#                                  "Tumour stage", "Tumour grade"))) |>
#   mutate(value = factor(value, levels = c("ER+", "ER-",
#                                           "PR+", "PR-",
#                                           "HER2-", "HER2+",
#                                           "N0", "N1-3",
#                                           "T1", "T2-4",
#                                           "Grade I/II", "Grade III",
#                                           "Unknown"))) |>
#   mutate(group = factor(group, levels = rev(c("low PRS,\nlow WID-qtBC", "high PRS,\nlow WID-qtBC", "low PRS,\nhigh WID-qtBC", "high PRS,\nhigh WID-qtBC"))))
# 
# text <- dat2 |>
#   select(value, group, set) |>
#   group_by(set, value, group) |>
#   summarise(n = n()) |>
#   mutate(group = factor(group, levels = c("low PRS,\nlow WID-qtBC", "high PRS,\nlow WID-qtBC", "low PRS,\nhigh WID-qtBC", "high PRS,\nhigh WID-qtBC"))) |>
#   mutate(freq = n / sum(n),
#          prop = rev(cumsum(rev(freq))))
# 
# dat3 <- dat2 |>
#   full_join(text, by = c("set", "group", "value"))
# 
# dat3$group <- gsub("WID-qtBC", "WID™-qtBC", dat3$group)
# dat3$group <- gsub("PRS", "PRS₃₁₃", dat3$group)
# 
# # Generate cols
# col2 <- rev(grDevices::colorRampPalette(col[c(3, 7, 6, 4)])(4))
# 
# fig3d <- dat3 |>
#   ggplot(aes(x = value,
#              fill = group,
#              label = n)) +
#   geom_bar(position = "fill",
#            alpha = 0.9) +
#   geom_text(size = 3, aes(y = (prop-(0.5*freq))),
#             check_overlap = T) +
#   xlab("") +
#   scale_fill_manual(values = col2,
#                     name = "") +
#   ylab("Proportion (%)") +
#   theme(axis.text.x = element_text(angle = 60),
#         panel.background = element_blank(),
#         legend.position = "top") +
#   facet_wrap(~set,
#              scales = "free_x",
#              nrow = 1) +
#   guides(fill = guide_legend(reverse = T))
# 
# # Linear model
# # x <- glm(group ~ stage_T + grade + ER + PR + HER2 + stage_N, data = dat, family = "binomial"(link="logit"))
# # summary(x)
# 
# plot <- (fig3a | fig3b | fig3c)/(fig3d) + plot_annotation(tag_levels = c("a")) +
#   plot_layout(heights = c(1, 0.8))
# 
# grDevices::cairo_pdf("2-markdown/0-figure-panels/fig3.pdf",
#                      width = 12, height = 9)
# print(plot)
# dev.off()
```

![](0-figure-panels/fig3.pdf)

\subsection{Figure 3. Association of the WID™-qtBC with the PRS\textsubscript{313} (313 SNP polygenic risk score) and cancer characteristics.} \textbf{a} Pearson correlation of the WID™-qtBC index with the polygenic risk score (PRS) in the Validation set. \textbf{b} ROC curve analysis of the WID™-qtBC index stratified by median PRS group (higher or lower than median) in the Validation set. \textbf{c} Odds ratio of the PRS\textsubscript{313}, WID™-qtBC, or their combination, across different percentile categories for the risk scores. Shading indicates 95% confidence intervals.  \textbf{d} Association of higher tumour stages with increased PRS313 and WID™-qtBC risk. Assessment of estrogen receptor (ER), progesterone receptor (PR), and human epidermal growth factor receptor 2 (HER2 status), nodal and tumour stage, and tumour grade amongst cases in the Validation set, comparing different risk groups defined in Table 1. Numbers in bars indicate n for each group. p=0.0349 for an association of T2-4 tumours with increasing PRS313 and WID™-qtBC, p=0.0167 for increasing Grade III tumours with increasing PRS313 and WID™-qtBC. p values were derived from logistic regression model using ER status, PR status, HER2 status, nodal stage, tumour stage, and tumour grade as independent variables and risk group as the dependent variable. ER status was not significant.


\newpage

```{r fig4}
# # Fig 4a (mifepristone before - after) ----------------
# load("1-analysis-pipeline/0-data/pheno.Rdata")
# pheno <- pheno |>
#   filter(dataset == "breast treatment set") |>
#   droplevels()
# 
# # Set colours and keep only paired
# df <- pheno |>
#   pivot_wider(id_cols = c(sample_id, type),
#               names_from = "time",
#               values_from = index_mQTL) |>
#   filter(!is.na(After))
# 
# df <- df |>
#   mutate(diff = After-Before,
#          col = ifelse(diff > 0, "WID™-qtBC increase", "WID™-qtBC reduction"))
# 
# ind <- match(pheno$sample_id, df$sample_id)
# pheno$col <- df$col[ind]
# 
# df2 <- df |>
#   group_by(type) |>
#   dplyr::summarise(p = paste("p = ", round(t.test(After, Before, paired = TRUE)$p.value, 2), sep = ""))
# 
# pheno <- pheno |>
#   filter(sample_id %in% df$sample_id)
# 
# pheno2 <- pheno |>
#   full_join(df2)
# 
# df2$x <- 1.5
# df2$y <- -2.5
# # df3 <- pheno |>
# #   group_by(type) |>
# #   do(tidy(t.test(index_mQTL ~ free_var1, data = ., paired = TRUE)))
# 
# fig4a <- pheno |>
#   dplyr::mutate(type = factor(type, levels = c("Vitamin", "Mifepristone"))) |>
#   ggplot(aes(x = time,
#              y = index_mQTL,
#              group = sample_id,
#              linetype = col)) +
#   geom_line() +
#   scale_linetype_manual(values = c("solid", "dashed"),
#                         name = "") +
#   # geom_point(size = 0.5) +
#   facet_wrap(~type) +
#   theme(legend.position = "top") +
#   xlab("") +
#   ylab("WID™-qtBC")
# 
# # Fig 4b association with age ----------------
# load("1-analysis-pipeline/0-data/pheno.Rdata")
# pheno <- pheno |>
#   filter(dataset == "validation set") |>
#   dplyr::mutate(type = factor(type, levels = c("Control", "Breast cancer"))) |>
#   droplevels()
# 
# fig4b <- pheno |>
#   ggplot(aes(x = age,
#              y = index_mQTL,
#              colour = type)) +
#   geom_point(size = 0.75,
#              alpha = 0.8) +
#   geom_smooth(method = "lm", se = F) +
#   ggpubr::stat_cor(size = 3,
#                    label.y.npc = "bottom") +
#   theme(legend.position = "top") +
#   scale_colour_manual(values = col[c(3,4)], name = "") +
#   ylab("WID™-qtBC")
# 
# # Fig 4c association with BMI ----------------
# fig4c <- pheno |>
#   ggplot(aes(x = BMI,
#              y = index_mQTL,
#              colour = type)) +
#   geom_point(size = 0.75,
#              alpha = 0.8) +
#   geom_smooth(method = "lm", se = F) +
#   ggpubr::stat_cor(size = 3,
#                    label.y.npc = "bottom") +
#   theme(legend.position = "none") +
#   scale_colour_manual(values = col[c(3,4)]) +
#   ylab("WID™-qtBC") +
#   xlab("body mass index")
# 
# # Fig 4d association with age at menarche ----------------
# fig4d <- pheno |>
#   ggplot(aes(x = age_at_menarche,
#              y = index_mQTL,
#              colour = type)) +
#   geom_point(size = 0.75,
#              alpha = 0.8) +
#   geom_smooth(method = "lm", se = F) +
#   ggpubr::stat_cor(size = 3,
#                    label.y.npc = "bottom") +
#   theme(legend.position = "none") +
#   scale_colour_manual(values = col[c(3,4)]) +
#   ylab("WID™-qtBC") +
#   xlab("age at menarche")
# 
# # Fig 4e association with age at menopause ----------------
# fig4e <- pheno |>
#   dplyr::filter(menopause == "Post") |>
#   ggplot(aes(x = age_at_menopause,
#              y = index_mQTL,
#              colour = type)) +
#   geom_point(size = 0.75,
#              alpha = 0.8) +
#   geom_smooth(method = "lm", se = F) +
#   ggpubr::stat_cor(size = 3,
#                    label.y.npc = "bottom") +
#   theme(legend.position = "none") +
#   scale_colour_manual(values = col[c(3,4)]) +
#   ylab("WID™-qtBC") +
#   xlab("age at menopause")
# 
# 
# design <- "
# AAB
# CDE
# "
# 
# plot <- (fig4a + fig4b + fig4c + fig4d + fig4e) + plot_annotation(tag_levels = "a") +
#   plot_layout(design = design)
# 
# grDevices::cairo_pdf("2-markdown/0-figure-panels/fig4.pdf",
#                      width = 9, height = 6.25)
# print(plot)
# dev.off()
```

![](0-figure-panels/fig4.pdf)


\subsection{Figure 4. Dynamic changes of the WID™-qtBC and association with non-genetic characteristics.} \textbf{a} Matched breast biopsy samples in healthy women before and after two months of vitamin or mifepristone treatment. 7/9 (77.8%) women in the mifepristone group showed a reduction in the WID™-qtBC while only 3/11 (27.3%) showed a reduction in the vitamin group.  Association of WID™-qtBC scores in cervical samples in the validation set with \textbf{b} age, \textbf{c} body mass index, \textbf{d} age at menarche, and \textbf{e} age at menopause in cancer cases and controls (R indicates Pearson correlation coefficient).

\newpage

\subsection{Table 1. Significantly increased risk of breast cancer for individuals with both a high PRS313 and high WID™-qtBC index.} Cut-offs were defined based on the median PRS\textsubscript{313} and WID™-qtBC (high scores were above the median). P values and odds ratios estimated using median-unbiased estimation. Statistically significant values are shown in bold.

```{r table.1}
# load("1-analysis-pipeline/0-data/pheno.Rdata")
# 
# # Discovery set
# disc <- pheno |>
#   filter(dataset == "discovery set" & !is.na(polygenic_risk_score_bc)) |>
#   mutate(type = factor(type, levels = c("Control", "Breast cancer"))) |>
#   droplevels() |>
#   mutate(grp = case_when(polygenic_risk_score_bc > median(polygenic_risk_score_bc) ~ "high PRS",
#                          polygenic_risk_score_bc <= median(polygenic_risk_score_bc) ~ "low PRS"),
#          grp = factor(grp, levels = c("low PRS", "high PRS")),
#          qtgrp = case_when(index_mQTL > median(index_mQTL) ~ "high WID-qtBC",
#                            index_mQTL <= median(index_mQTL) ~ "low WID-qtBC"),
#          qtgrp = factor(qtgrp, levels = c("low WID-qtBC", "high WID-qtBC"))) |>
#   tidyr::unite("group", grp:qtgrp, sep = ", ", remove = FALSE) |>
#   mutate(group = factor(group, levels = c("low PRS, low WID-qtBC",
#                                           "high PRS, low WID-qtBC",
#                                           "low PRS, high WID-qtBC",
#                                           "high PRS, high WID-qtBC"))) |>
#   mutate(type = factor(type, levels = c("Control", "Breast cancer")))
# disc_data <- ortable(disc$type, disc$group)
# disc_data$set <- "Discovery set"
# 
# val <- pheno |>
#   filter(dataset == "validation set" & !is.na(polygenic_risk_score_bc)) |>
#   mutate(type = factor(type, levels = c("Control", "Breast cancer"))) |>
#   droplevels() |>
#   mutate(grp = case_when(polygenic_risk_score_bc > median(polygenic_risk_score_bc) ~ "high PRS",
#                          polygenic_risk_score_bc <= median(polygenic_risk_score_bc) ~ "low PRS"),
#          grp = factor(grp, levels = c("low PRS", "high PRS")),
#          qtgrp = case_when(index_mQTL > median(index_mQTL) ~ "high WID-qtBC",
#                            index_mQTL <= median(index_mQTL) ~ "low WID-qtBC"),
#          qtgrp = factor(qtgrp, levels = c("low WID-qtBC", "high WID-qtBC"))) |>
#   tidyr::unite("group", grp:qtgrp, sep = ", ", remove = FALSE) |>
#   mutate(group = factor(group, levels = c("low PRS, low WID-qtBC",
#                                           "high PRS, low WID-qtBC",
#                                           "low PRS, high WID-qtBC",
#                                           "high PRS, high WID-qtBC"))) |>
#   mutate(type = factor(type, levels = c("Control", "Breast cancer")))
# val_data <- ortable(val$type, val$group)
# val_data$set <- "Validation set"
# 
# data <- rbind(disc_data, val_data)
# data$group <- gsub("WID-qtBC", "WID™-qtBC", data$group)
# data$group <- gsub("PRS,", "PRS₃₁₃,", data$group)
# data$p <- as.numeric(data$p)
# 
# tab <- data |>
#   gt(groupname_col = "set") |>
#   text_transform(locations = cells_body(
#     columns = p
#   ),
#                  fn = function(x){
#                    paste0("p=", x)}
#   ) |>
#   text_transform(locations = cells_body(),
#                  fn = function(x){
#                    paste0(ifelse(x == "p=NA", "-",  x))}
#   ) |>
#   cols_label(`group` = "Risk group",
#              OR = "Odds ratio",
#              `95% CI` = "95% CI",
#              p = "p value",
#              Controls = "Controls (n)",
#              Cases = "Cases (n)") |>
#   cols_align(columns = c(Cases,
#                          Controls,
#                          OR,
#                          `95% CI`,
#                          p),
#              align = "left") |>
#   tab_style(
#     style = cell_text(weight = "bold"),
#     locations = cells_body(
#       columns = c(OR,
#                   `95% CI`,
#                   p),
#       rows = p < 0.05)
#   ) |>
#   tab_options(column_labels.font.weight = "bold",
#               row_group.font.weight = "bold",
#               data_row.padding = 2,
#               column_labels.font.size = 12,
#               table.font.size = 12,
#               row_group.padding = 2,
#               row_group.border.right.width = px(10),
#               summary_row.padding = 2,
#               table.width = px(700),
#               table.border.top.color = "white",
#               row_group.border.top.width = px(1),
#               row_group.border.bottom.width = px(1),
#               stub.border.width = px(0),
#               heading.title.font.size = 14) |>
# gt::tab_source_note(html("<b>Low</b> and <b>high</b> grouping was defined on median values of the respective risk score in each set. <b>High</b> scores were above the median, while <b>low</b> scores were defined as equal to or below the median.")) |> 
#   opt_stylize(style = 3)
# 
# 
# tab |>
#   gtsave("2-markdown/0-figure-panels/table-1.html")
# 
# # Saving html to pdf:
# pagedown::chrome_print("2-markdown/0-figure-panels/table-1.html", output = "2-markdown/0-figure-panels/table-1.pdf")
```

\includegraphics[trim=0 5cm 0 0]{0-figure-panels/table-1.pdf}
